aiida_common_workflows/workflows/dissociation.py

```diff
@@ -20,6 +20,15 @@ def validate_inputs(value, _):
         if value['distance_min'] >= value['distance_max']:
             return '`distance_min` must be smaller than `distance_max`'
 
+    # Validate that the provided ``generator_inputs`` are valid for the associated input generator.
+    process_class = WorkflowFactory(value['sub_process_class'])
+    generator = process_class.get_input_generator()
+
+    try:
+        generator.get_builder(value['structure'], **value['generator_inputs'])
```

Comment:
```suggestion
        generator.get_builder(value['molecule'], **value['generator_inputs'])
```

Reply:
Good catch. I was wondering why this wasn't failing the tests though, because this should have raised a `KeyError`. The reason is that in the tests I had the same copy :spaghetti: error and also passed `structure`. But then again that *also* should raise as that is not a valid port. But *then* I realized that the `generator_inputs` namespace is *dynamic* and so accepts anything (silently). These namespaces really shouldn't be dynamic, so I removed those from both DC and EOS

Reply:
Nice 👍 . I didn't even know that namespaces could be dynamic. 😅